documentation/asciidoc/computers/raspberry-pi/spi-bus-on-raspberry-pi.adoc

```diff
@@ -1,104 +1,299 @@
 [[spi-overview]]
 == Serial Peripheral Interface (SPI)
 
-The Raspberry Pi family of devices is equipped with a number of https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus[SPI] buses. SPI can be used to connect a wide variety of peripherals - displays, network controllers (Ethernet, CAN bus), UARTs, etc. These devices are best supported by kernel device drivers, but the `spidev` API allows userspace drivers to be written in a wide array of languages.
+Raspberry Pi computers are equipped with a number of https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus[SPI] buses. SPI can be used to connect a wide variety of peripherals - displays, network controllers (Ethernet, CAN bus), UARTs, etc. These devices are best supported by kernel device drivers, but the `spidev` API allows userspace drivers to be written in a wide array of languages.
 
 [[spi-hardware]]
 === SPI Hardware
 
-The BCM2835 core common to all Raspberry Pi devices has 3 SPI Controllers:
+The BCM2835 core common to all Raspberry Pi computers has 3 SPI controllers:
```

Comment:
This probably needs additional rewording still? BCM2835 -> VideoCore IV? (since Pis 2 and 3 have BCM2836 and BCM2837.) And "all Raspberry Pi computers prior to the Raspberry Pi 4" ?

Reply:
> This probably needs additional rewording still? BCM2835 -> VideoCore IV? (since Pis 2 and 3 have BCM2836 and BCM2837.) And "all Raspberry Pi computers prior to the Raspberry Pi 4" ?

I've rewritten this. I've also got a question in my mind as to whether CM4 has SPI1 available on it, since there's less GPIO pins exposed on CM4 compared to Pi 4.

Reply:
> I've also got a question in my mind as to whether CM4 has SPI1 available on it

ping @dp111

Reply:
The public datasheets also don't specify where SPI2 is.

Reply:
https://datasheets.raspberrypi.org/bcm2711/bcm2711-peripherals.pdf says "SPI 2 is not user-accessible" so it appears this affects all BCM2711 boards, not just CM4!

Reply:
> https://datasheets.raspberrypi.org/bcm2711/bcm2711-peripherals.pdf says "SPI 2 is not user-accessible" so it appears this affects all BCM2711 boards, not just CM4!

I've changed the reference to SPI2 on this page to mention that it is only accessible on CM1, 3, 3+. Previously it just said `compute modules` which implied all models of CM.